manager/orchestrator/replicated/services.go

```diff
@@ -198,10 +198,23 @@ func (r *Orchestrator) addTasks(ctx context.Context, batch *store.Batch, service
 	}
 }
 
-func (r *Orchestrator) deleteTasks(ctx context.Context, batch *store.Batch, slots []orchestrator.Slot) {
+// setTasksDesiredState sets the desired state of all tasks to what
+// is requested.
+func (r *Orchestrator) setTasksDesiredState(ctx context.Context, batch *store.Batch, slots []orchestrator.Slot, state api.TaskState) {
 	for _, slot := range slots {
 		for _, t := range slot {
-			r.deleteTask(ctx, batch, t)
+			err := batch.Update(func(tx store.Tx) error {
```

Comment:
Why not do all the updates in the same txn ?

Reply:
@anshulpundir how do you mean?

Reply:
Nevermind. I realized that updates to the store are batched using store.Batch.